test/common/websocket/codec_test.cc

```diff
@@ -0,0 +1,824 @@
+#include <array>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/websocket/codec.h"
+
+#include "test/common/buffer/utility.h"
+#include "test/test_common/printers.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace WebSocket {
+namespace {
+
+// A single-frame unmasked text message
+// 0x81 0x05 0x48 0x65 0x6c 0x6c 0x6f (contains "Hello")
+//                   H        e        l        l        o
+// 10000001 00000101 01001000 01100101 01101100 01101100 01101111
+// fin - 1
+// opcode - 0x1 - text data
+// mask bit - 0
+// length - 5 bytes 0000101
+TEST(WebSocketCodecTest, decodeSingleUnmaskedFrame) {
```

Comment:
nit: the 2nd argument to TEST() should start with a Capital:
```
TEST(WebSocketCodecTest, DecodeSingleUnmaskedFrame) {
```
Please apply here and elsewhere.